app/Http/Controllers/ControllerDash.php

```diff
@@ -354,6 +355,9 @@ public function showEvents() {
                 return strtotime($e->date);
             });
         }
+        foreach ($events as $e) {
+            $this->reduceEventBanner($e);
+        }
```

Comment:
I would recommend handling this logic in a command that runs daily. If we run this every time someone goes to the event page, this could get very, very slow.

Reply:
The full method to reduce the banner doesn't run every time. If a reduced banner has already been created (found in the directory), then it uses the existing reduced-size banner. It will only create a new reduced banner if one does not exist. I prefer this method over it running once daily (although I did consider that and landed on this method). The reasoning is that if the EC uploads a new banner 1 minute after our daily routine, then anyone that accesses the website from then until 23 hours and 59 minutes later will have to download the full-res banner. With this method, there is never an opportunity for the full-res banner to get served back to a user, and the only user that has to pay a slight load time penalty is the first one to see it (which will normally be the EC).

Reply:
The other thing that I could do would be to add the reduced banner image link to the database. That might be significantly faster than the directory search, but then I wouldn't be assured that the image exists before displaying it... oh, the trade-offs

Reply:
Ah that's a good point since the EC will normally be the first to see it... how about we reduce it when the event is being saved? Then we don't have to worry about load times and ensure that the EC is the one that sees any type of performance reduction?

Reply:
Ok, here's the resolution:
1. Removed reduceEventBanner() calls from both FrontController and ControllerDash. This is now called when an event is created or edited as part of the save method. Added a column to the database that stores the reduced banner path.
2. Created a new method displayBannerPath() that is called in FrontController and ControllerDash that determines which banner (full res or reduced res) to display based on a file exists call. This ensures that if one of the banners is deleted or not available (not sure why this would happen, but...) the user still sees an appropriate image.